src/reporter/github.test.js

```diff
@@ -0,0 +1,519 @@
+        .post(`/repos/owner/repo/issues/${issueNumber}/comments`, { body: comment })
+        .reply(200);
+
+      await github.addCommentToIssue({ issue: { number: issueNumber }, comment });
+    });
+
+    after(nock.cleanAll);
+
+    it('adds the comment to the issue', async () => {
+      expect(scope.isDone()).to.be.true;
+    });
+  });
+
+  describe('#closeIssueWithCommentIfExists', () => {
+    after(nock.cleanAll);
+
+    context('when the issue exists and is open', () => {
+      let issue;
+      let getIssuesScope;
+      let addCommentScope;
+      let closeIssueScope;
+
+      before(async () => {
+        issue = { number: 123, title: 'Open Issue', state: GitHub.ISSUE_STATE_OPEN };
+
+        getIssuesScope = nock('https://api.github.com')
+          .get('/repos/owner/repo/issues')
+          .query(true)
+          .reply(200, [issue]);
+
+        addCommentScope = nock('https://api.github.com')
+          .post(`/repos/owner/repo/issues/${issue.number}/comments`)
+          .reply(200);
+
+        closeIssueScope = nock('https://api.github.com')
+          .patch(`/repos/owner/repo/issues/${issue.number}`, { state: GitHub.ISSUE_STATE_CLOSED })
+          .reply(200);
+
+        await github.closeIssueWithCommentIfExists({ title: issue.title, comment: 'Closing comment' });
+      });
+
+      it('searches for the issue', () => {
+        expect(getIssuesScope.isDone()).to.be.true;
+      });
+
+      it('adds comment to the issue', () => {
+        expect(addCommentScope.isDone()).to.be.true;
+      });
+
+      it('closes the issue', () => {
+        expect(closeIssueScope.isDone()).to.be.true;
+      });
+    });
+
+    context('when the issue exists and is closed', () => {
+      let issue;
+      let getIssuesScope;
+      let addCommentScope;
+      let closeIssueScope;
+
+      before(async () => {
+        issue = { number: 123, title: 'Closed Issue', state: GitHub.ISSUE_STATE_CLOSED };
+
+        getIssuesScope = nock('https://api.github.com')
+          .get('/repos/owner/repo/issues')
+          .query(true)
+          .reply(200, []);
+
+        addCommentScope = nock('https://api.github.com')
+          .post(`/repos/owner/repo/issues/${issue.number}/comments`)
+          .reply(200);
+
+        closeIssueScope = nock('https://api.github.com')
+          .patch(`/repos/owner/repo/issues/${issue.number}`, { state: GitHub.ISSUE_STATE_CLOSED })
+          .reply(200);
+
+        await github.closeIssueWithCommentIfExists({ title: issue.title, comment: 'Closing comment' });
+      });
+
+      it('searches for the issue', () => {
+        expect(getIssuesScope.isDone()).to.be.true;
+      });
+
+      it('does not attempt to add comment', () => {
+        expect(addCommentScope.isDone()).to.be.false;
+      });
+
+      it('does not attempt to close the issue', () => {
+        expect(closeIssueScope.isDone()).to.be.false;
+      });
+    });
+
+    context('when the issue does not exist', () => {
```

Comment:
While I am convinced of the relevance of testing this case, I am not sure of the relevance of the suggested tests. What does it mean to test that we do not try to add a comment to nor close a non-existent issue? If anything, we should check that our code gracefully handles non-200 replies, but checking that we do not send back further requests does not seem relevant to me.

Reply:
In fact, if the issue does not exist we still receive a 200 HTTP code with an empty result `[]`. If we try to add a comment or close the issue, we probably have an error in our code.